File: backend/src/main/java/com/automobil/backend/dto/FormAdvert.java
package com.automobil.backend.dto;
import lombok.Data;
import org.springframework.web.multipart.MultipartFile;


import java.util.Date;
import java.util.List;

@Data
public class FormAdvert {
    private String carbodyTitle;
    private Long clientsId;
    private String marksTitle;
    private String modelTitle;
    private Long generationsId;
    private Long modificationsId;
    private String available;
    private Double price;
    private String colour;
    private Double mileage;
    private Integer broken;
    private String city;
    private String place;
    private Long phone;
    private Integer swap;
    private Integer pts;
    private Integer owners;
    private String gosnumber;
    private String buydays;
    private String dayofwarrantys;
    private String vin;
    private Long sts;
    private String commentns;
    private List<MultipartFile> files;
}

File: frontend/src/app/model/formAdvert.ts
import {Client} from './client';
import {Generation} from './generation';
import {Modification} from './modification';
import {Mark} from './mark';
import {Model} from './model';

export class FormAdvert{
  carbodyTitle: string;
  clientId: bigint;
  mark: Mark;
  model: Model;
  generations: Generation;
  modifications: Modification;
  available: string;
  price: number;
  colour: string;
  mileage: number;
  broken: boolean;
  city: string;
  place: string;
  phone: bigint;
  swap: boolean;
  pts: boolean;
  owners: number;
  gosnumber: string;
  buyday: number;
  dayofwarranty: Date;
  vin: string;
  sts: bigint;
  commentns: string;
  files: File[]=[];

}
